### PR TITLE
[top_darjeeling,lint,sim] Add lint and sim cfg for Darjeeling

### PR DESCRIPTION
**Pull request by Razer6**
This PR adds the block level lint and sim configuration for the Darjeeling top. It only adds the IPs that are used in Darjeeling, ignoring other IPs that are used in Earlgrey.

The PR enables the DMA, MBX, KeymgrDPE, soc_proxy IP.

### CODE REVIEW

**Comment by Razer6** [comment]
I see that those are not yet part of CI - should we already enable the configs for the other IP?

File: hw/top_darjeeling/lint/top_darjeeling_dv_lint_cfgs.hjson
Comment (rswarbrick) [inline hw/top_darjeeling/lint/top_darjeeling_dv_lint_cfgs.hjson:26]: Would it make sense to open an issue about the comment characters in this list so that we can't accidentally forget to turn on these lint configurations once we've merged the blocks / turned on the top-level autogen ones?

**Comment by rswarbrick** [comment]
About the CI question: yes, probably :-) At the moment, the CI version of the lint flow runs through `ci/scripts/verible-lint.sh`, which just knows about Earlgrey. I think the way to do it properly would be to add an extra "toplevel" argument to the script and then use that when constructing the hjson paths. At the moment, we run the script three times (RTL, DV, FPV) in `slow-lint.sh`. Splitting it out like that, we'd have to bump that up to three times for each top-level.

File: hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson
Comment (rswarbrick) [inline hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson:32]: Nit: Indentation is off here, I think.

File: hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson
Comment (rswarbrick) [inline hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson:22]: Hmm: do we have any way to do this yet? I think the Earlgrey version comes from `hw/top_earlgrey/ip_autogen/alert_handler/fpv`. I think this will probably need commenting until the Darjeeling equivalent lands?

Similarly for the ping timer.

File: hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson
Comment (rswarbrick) [inline hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson:38]: This seems a little silly to me. All the blocks below are identical between the top-levels (I think) and they probably shouldn't have been put in a file with "top_earlgrey" in the name before. Can we split things out into some "shared_fpv_lint_cfgs" list? (Maybe as a separate PR that lands before this one)

File: hw/top_darjeeling/lint/top_darjeeling_lint_cfgs.hjson
Comment (rswarbrick) [inline hw/top_darjeeling/lint/top_darjeeling_lint_cfgs.hjson:40]: I don't think this can land until the corresponding autogenerated IP is in, can it? Maybe this item needs commenting? Similarly for quite a few other blocks below.

File: hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson
Comment (Razer6) [reply hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson:38]: Would adding that to the `import_cfgs` section work?

Comment (rswarbrick) [reply hw/top_darjeeling/lint/top_darjeeling_fpv_lint_cfgs.hjson:38]: I genuinely don't know! (And if it doesn't work, that might be a bug we should fix in dvsim...)

File: hw/top_darjeeling/lint/top_darjeeling_dv_lint_cfgs.hjson
Comment (Razer6) [reply hw/top_darjeeling/lint/top_darjeeling_dv_lint_cfgs.hjson]: Done in https://github.com/lowRISC/opentitan/issues/24737

**Review by rswarbrick — APPROVED** [review]
This looks sensible to me, and thanks for splitting things out like this.

**Comment by rswarbrick** [comment]
Oh, actually it looks like something has gone slightly wrong with the includes here:
image

**Comment by Razer6** [comment]
@rswarbrick I'm not sure if nested configs are supported by the flow (nested in the sense of more than one child config)

**Comment by rswarbrick** [comment]
Oh, that's a pity. But the change in the latest version (doing the expansion manually) looks like a reasonable approach for now, at least. Thanks.

**Comment by Razer6** [comment]
Yep. Wanted to get the lint in, to unblock the other PRs.

**Comment by Razer6** [comment]
@rswarbrick CI is green - can we get that merged?

**Pull request merged**
